// extensions/index.js
//returns the string definition of a card type
function GetCardType(cardType) {
    switch (cardType) {
        case 1:
            return "monster";
        case 2:
            return "spell";
        case 3:
            return "trap";
    }
}

//a function to turn strings from xyz -> Xyz
function toTitleCase() {
    return this.replace(/\w\S*/g, function (txt) {
        return txt.charAt(0).toUpperCase() + txt.substr(1).toLowerCase();
    });
};

//returns true if the target array contains any items of the param array
function contains(array2) {
    for (var i = 0; i < this.length; i++) {
        for (var j = 0; j < array2.length; j++) {
            if (this[i] == array2[j])
                return true;
        }
    }
    return false;
};

module.exports = {
    GetCardType,
    extendArray: function (constructor) {
        constructor.prototype.contains = contains;
    },
    extendString: function (constructor) {
        constructor.prototype.contains = contains;
        constructor.prototype.toTitleCase = toTitleCase;
    }
};
